iceoryx_hoofs/buffer/include/iox/stack.hpp

```diff
@@ -34,6 +35,9 @@ template <typename T, uint64_t Capacity>
 class stack final // NOLINT(cppcoreguidelines-pro-type-member-init, hicpp-member-init)
 {
   public:
+    // Due to 'pop' returning an 'iox::optional'
+    static_assert(std::is_move_constructible<T>::value, "T has to be move constructible");
```

Comment:
Could you please move the condition to an external constexpr variable so that we can check in different instances if the type is compatible with the stack beforehand. If it is directly in here we have no other way than fail instead of providing a nice message.

So:
```cxx
template<typename T>
constexpr bool TYPE_FULFILLS_STACK_REQUIREMENTS = std::is_move_constructible<T>::value;

// this line in the stack

static_assert(TYPE_FULFILLS_STACK_REQUIREMENTS<T>, "..."); 
```

Now we can ask in other templates with `TYPE_FULFILLLS_STACK_REQUIREMENTS` if the type works with a stack.

Reply:
@elfenpiff Got your point. What I don't like that too much about that we'll end up with inconsistent naming of helper templates, I'm really missing the formality of C++20 concepts here :pensive: Right now, most of the time the `static_assert`s are in the class e.g. `CyclicIndex`, `SoFi`, `PolymorphicHandler`, etc.

I would stick with the idiomatic C++ way of naming the helper template if you don't mind:

```cpp
template <typename T>
using is_storable_in_iox_stack_t = typename std::is_move_constructible<T>::type;

// the following line in the stack

static_assert(is_storable_in_iox_stack_t<T>::value, "T has to be move constructible");
```

Does that make sense?

Reply:
@mossmaurice Good idea. Let's do it in your proposed way.